src/test/java/co/rsk/federate/FedNodeRunnerTest.java

```diff
@@ -0,0 +1,219 @@
+package co.rsk.federate;
+
+import co.rsk.NodeRunner;
+import co.rsk.bitcoinj.core.NetworkParameters;
+import co.rsk.config.BridgeConstants;
+import co.rsk.federate.btcreleaseclient.BtcReleaseClient;
+import co.rsk.federate.config.FedNodeSystemProperties;
+import co.rsk.federate.config.SignerConfig;
+import co.rsk.federate.log.FederateLogger;
+import co.rsk.federate.log.RskLogMonitor;
+import co.rsk.federate.signing.ECDSACompositeSigner;
+import co.rsk.federate.signing.ECDSASigner;
+import co.rsk.federate.signing.hsm.advanceblockchain.HSMBookkeepingService;
+import co.rsk.federate.signing.hsm.client.HSMBookkeepingClient;
+import co.rsk.federate.signing.utils.TestUtils;
+import com.typesafe.config.Config;
+import org.ethereum.config.Constants;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Created by Kelvin Isievwore on 24/04/2023.
+ */
+public class FedNodeRunnerTest {
+    private FedNodeRunner fedNodeRunner;
+    private FedNodeSystemProperties fedNodeSystemProperties;
+    private Config config;
+
+    @Before
+    public void setUp() {
+        BridgeConstants bridgeConstants = mock(BridgeConstants.class);
+        Constants constants = mock(Constants.class);
+        fedNodeSystemProperties = mock(FedNodeSystemProperties.class);
+        when(fedNodeSystemProperties.getNetworkConstants()).thenReturn(constants);
+        when(constants.getBridgeConstants()).thenReturn(bridgeConstants);
+        when(bridgeConstants.getBtcParamsString()).thenReturn(NetworkParameters.ID_REGTEST);
+
+        config = mock(Config.class);
+        when(config.getString("host")).thenReturn("127.0.0.1");
+        when(config.getInt("port")).thenReturn(9999);
```

Comment:
Are these params needed when using keyfile?

Reply:
Only Path Is Needed, Removed The Rest